packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -272,82 +116,62 @@ const generateComponentMeta = (): void => {
       ...(source.match(/<StateMessage/) ? ['p-icon' as TagName] : []),
     ].filter((x, idx, arr) => arr.findIndex((t) => t === x) === idx); // remove duplicates;
 
-    const deprecatedProps: ComponentMeta['deprecatedProps'] = [];
-
-    const arrayProps: ComponentMeta['arrayProps'] = [];
-
     // props
     const propsMeta: ComponentMeta['propsMeta'] = {};
 
-    const props: ComponentMeta['props'] = Array.from(
+    Array.from(
       source.matchAll(/(  \/\*\*[\s\S]+?)?@Prop\(.*\) public ([a-zA-Z]+)\??(?:(?:: (.+?))| )(?:=[^>]\s*([\s\S]+?))?;/g)
-    ).reduce(
-      (result, [, jsdoc, propName, propType, propValue]) => {
-        let cleanedValue: boolean | number | string | object =
-          propValue === 'true'
-            ? true
-            : propValue === 'false'
-              ? false
-              : // undefined values get lost in JSON.stringify, but null is allowed
-                propValue
-                  ?.replace(/^['"](.*)['"]$/, '$1') // propValue is a string and might contain a string wrapped in quotes since it is extracted like this
-                  .replace(/\s+/g, ' ') // remove new lines and multiple spaces
-                  .replace(/,( })/, '$1') ?? null; // remove trailing comma in original multiline objects
-
-        if (typeof cleanedValue === 'string') {
-          if (cleanedValue.match(/^\d+$/) && !propValue.match(/^['"]\d+['"]$/)) {
-            // parse numbers
-            cleanedValue = parseInt(cleanedValue);
-          } else if (cleanedValue.match(/^{.+}$/)) {
-            // parse objects
-            cleanedValue = eval(`(${cleanedValue})`);
-          } else if (cleanedValue.match(/\[.*]/g)) {
-            // parse arrays
-            if (cleanedValue !== '[]') {
-              // TODO: Support non empty array values
-              throw new Error(`Expected an empty array '[]' for prop '${propName}', but found '${propValue}'`);
-            }
-            arrayProps.push(propName);
-            cleanedValue = [];
+    ).forEach(([, jsdoc, propName, propType, propValue]) => {
+      let cleanedValue: boolean | number | string | object =
+        propValue === 'true'
+          ? true
+          : propValue === 'false'
+            ? false
+            : // undefined values get lost in JSON.stringify, but null is allowed
+              propValue
+                ?.replace(/^['"](.*)['"]$/, '$1') // propValue is a string and might contain a string wrapped in quotes since it is extracted like this
+                .replace(/\s+/g, ' ') // remove new lines and multiple spaces
+                .replace(/,( })/, '$1') ?? null; // remove trailing comma in original multiline objects
```

Comment:
I know it was like this before but this is unreadable & unmaintainable. You should consider refactoring it.